### DISCUSSION TITLE
merge two files Yaml

### DISCUSSION THREAD

**Discussion by anouarbouayad** [mikefarah/yq#1713] [Q&A]
I want to merge two yaml files file_dest.yml and file_source.yml. 
"file_dest.yml"
---
type: PortalConfiguration
fields:
  apiregAuthenticationPolicies: /Policies/YAML_PROJECT_120623/SECU_YAML_PROJECT_120623

"file_source.yml"
---
type: PortalConfiguration
fields:
  apiregAuthenticationPolicies:
    - /Policies/COMMUN/SECURITE/API_KEY/SECU_APIKEY_SECRET
    - /Policies/BIG/SECURITY_POLICIES/SECU_BIG
    
I use the following query:
yq eval-all 'select(fileIndex == 0) *+ select(fileIndex == 1)' "file_dest.yml" "file_source.yml"
The result of this request is 
---
type: PortalConfiguration
fields:
  apiregAuthenticationPolicies: /Policies/YAML_PROJECT_120623/SECU_YAML_PROJECT_120623

I want to produce the following file the merge of two files.

---
type: PortalConfiguration
fields:
  apiregAuthenticationPolicies:
    - /Policies/COMMUN/SECURITE/API_KEY/SECU_APIKEY_SECRET
    - /Policies/BIG/SECURITY_POLICIES/SECU_BIG
    - /Policies/YAML_PROJECT_120623/SECU_YAML_PROJECT_120623

Help me

**Comment by mikefarah** [comment]
Sorry missed this. yq has merge flags that can be used, one of which specifies to merge arrays `*+` see here: https://mikefarah.gitbook.io/yq/operators/multiply-merge.

However, there is an issue - the first file `apiregAuthenticationPolicies` is a string value whereas the second file its an array of strings - so the normal merge process won't work.

Best thing I can think to do is use yq to change it to an array, and then merge them:

```
yq ea '(.apiregAuthenticationPolicies |= [] + . ) | select(fi==0) *+ select(fi==1)' file1.yaml file2.yaml
```

Explanation:
- Use `ea` (or eval-all) to load both files into memory (as opposed to processing each one independently)
- `(.apiregAuthenticationPolicies |= [] + . )` will update `apiregAuthenticationPolicies` of both files to be an array if it isn't already one
- `select(fi==0) *+ select(fi==1)` will merge the first file (fi==0) with the second file. `*=` tells it to append any arrays see https://mikefarah.gitbook.io/yq/operators/multiply-merge for more information.
